drafts/25-xxx-specifications.txt

```diff
@@ -99,68 +98,107 @@ li32. Fortran source fragments may be continued with a continuation
+------------------------------
 
 ws01. The whitespace characters blank and horizontal tab character may appear
       on directive lines.
 
 ws03. Whitespace characters are significant in determining token
       boundaries in preprocessor directive lines.
 
-ws02. Whitespace characters are significant in determining token
+ws05. Outside of character constants, multiple whitespace characters
+      on a directive line are treated as a single space.
+
+ws07. Whitespace characters are significant in determining token
       boundaries for the purposes of recognizing macro names,
       in both fixed-form and free-form Fortran source fragments.
 
-ws03. Whitespace characters are significant in determining token
+ws09. Whitespace characters are significant in determining token
       boundaries for the purposes of recognizing macro names,
       in both fixed-form and free-form Fortran comment lines.
 
-ws04. In fixed-form input, macro names are not recognized as such when
+ws11. In fixed-form input, macro names are not recognized as such when
       spaces are inserted into their invocations.
 
 
-2.2 Comments
+2.4 Comments
 ------------
 
 co01. Directive lines may contain C-style '/*' ... '*/' comments.
 
-co02. Directive lines shall not contain C style '//' comments.
+co03. Directive lines shall not contain C style '//' comments.
 
-co02. '/*' ... '*/' comments on directive lines shall extend past a
+co05. '/*' ... '*/' comments on directive lines shall extend past a
       new-line only if the line ends in '\' new-line, indicating a
       continuation line. (Maybe controversial)
 
-co03. '/*' ... '*/' comments on directive lines are replaced by a
+co07. '/*' ... '*/' comments on directive lines are replaced by a
       single space, as specified in C 2023 5.1.1.2 bullet 3.
 
-co04. In a directive line, the '!' character is not interpreted as introducing
+co09. In a directive line, the '!' character is not interpreted as introducing
       a Fortran-style comment, and neither the `!` character nor any subsequent
       text are removed by the preprocessor.
 
-co05. Directive lines (by definition) cannot contain Fortran
+co11. Directive lines (by definition) cannot contain Fortran
       fixed-form 'C' or '*' style comments.
 
 
-2.4 Token lexicon
+2.5 Token lexicon
 -----------------
-[gak: List the character sequences that are recognized as separate tokens
-on directives and in Fortran source fragments. This should include
-Fortran identifiers, Fortran numeric constants, Fortran character
-constants, the Fortran operators, and the C operators used to construct
-constant-integer-expression.]
+
+The preprocessor is a token-manipulation engine (See C 2023 5.1.1.2
+Translation phases).
+As such, there is a specific lexicon of tokens recognized by
+the preprocessor (including unrecognizable tokens).
+
+We use illustrative syntax to describe the directive specifications,
+and the execution behavior of the preprocess on Fortran comment lines
```

Comment:
I continue to strongly oppose misusing the term "execution" to refer to any part of compilation.

I believe the C spec exclusively uses the term "execution" to describe program execution. Any actions taken by the preprocessor or compiler are "translation" (see C 23 5.0-1). Based on a quick skim, the same appears to be true of the Fortran spec.

```suggestion
and the translation behavior of the preprocess on Fortran comment lines
```